src/WebJobs.Extensions.DurableTask/ContextImplementations/RemoteOrchestratorContext.cs

```diff
@@ -73,6 +74,18 @@ internal bool TryGetOrchestrationErrorDetails(out Exception? failure)
 
         internal void SetResult(IEnumerable<OrchestratorAction> actions, string customStatus)
         {
+            // Azure Table Storage enforces a 32 KB limit if a property value is a UTF-16 encoded string.
+            // We apply a 16 KB limit here to align with our in-process model.
+            const int maxCustomStatusSizeInKB = 16;
+            int? customStatusSizeInKB = customStatus != null ?
```

Comment:
Suggestion: you could make this logic a bit simpler by using `double` instead of `int?`. It will also fix a precision bug where 16.9 KB gets converted to 16 KB as part of the cast to `int`.
```csharp
const int MaxCustomStatusSizeInKB = 16;

string customStatus = "hi";
double customStatusSizeInKB = customStatus != null ? Encoding.Unicode.GetByteCount(customStatus) / 1024.0 : 0;
	
if (customStatusSizeInKB > MaxCustomStatusSizeInKB)
{
	throw new ArgumentException(
	    $"CustomStatus too large: limit = {MaxCustomStatusSizeInKB} KB (UTF-16), actual = {customStatusSizeInKB:N2} KB.");
}
```
Then you don't have to check for `.HasValue` below.